src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -66,18 +66,26 @@ function getClauseForType(
   namespace: string | undefined,
   type: string
 ) {
-  if (namespace && !registry.isNamespaceAgnostic(type)) {
+  if (registry.isMultiNamespace(type)) {
+    return {
+      bool: {
+        must: [{ term: { type } }, { term: { namespaces: namespace ?? 'default' } }],
+        must_not: [{ exists: { field: 'namespace' } }],
+      },
+    };
+  } else if (namespace && registry.isSingleNamespace(type)) {
     return {
       bool: {
         must: [{ term: { type } }, { term: { namespace } }],
+        must_not: [{ exists: { field: 'namespaces' } }],
       },
     };
   }
-
+  // isNamespaceAgnostic
```

Comment:
nit: could also be single namespace in the default space

Reply:
Good point, I'll clarify the comment